= This is dirty but protects against a bunch of possible kernel panics caused
= by malicous libgeom requests. They only need access to /dev/geom.ctl, ie
= be in the operator group. Not necessarily root.
=
= THIS IS NOT SUPPOSED TO BE PUSHED UPSTREAM.
= Instead they should fix the calls to gctl_error and not throw away the error
= status... and not access kvalue when GCTL_PARAM_KERNVALUE isn't set... etc.

diff --git a/sys/geom/geom_ctl.c b/sys/geom/geom_ctl.c
index d8ca92f..48d310e 100644
--- a/sys/geom/geom_ctl.c
+++ b/sys/geom/geom_ctl.c
@@ -102,6 +102,7 @@ gctl_error(struct gctl_req *req, const char *fmt, ...)
 	}
 	if (req->nerror)
 		return (req->nerror);
+	req->nerror = EINVAL;
 
 	va_start(ap, fmt);
 	sbuf_vprintf(req->serror, fmt, ap);
